### ISSUE TITLE
Document.save() owner_pw & user_pw has buffer overflow bug.

### ISSUE THREAD

**Issue by lys0829** [pymupdf/PyMuPDF#2086]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
owner_pw & user_pw arguments in Document.save()has buffer overflow bug. May cause program crash and arbitrary code execution.

## To Reproduce (mandatory)
PoC:
```
import fitz

pdf = fitz.Document()
pdf.new_page()
pdf.save(1, owner_pw="a"*1000)
```

## Your configuration (mandatory)
```
FROM ubuntu:20.04
MAINTAINER Lys

RUN apt-get update && \
    DEBIAN_FRONTEND=noninteractive apt-get install -yq python3.8=3.8.10-0ubuntu1~20.04.5 python3-pip

RUN pip3 install PyMuPDF==1.20.2
```

**Comment by JorjMcKie** [comment]
Thanks for the report.
There exists a limitation to 40 characters. We will document this and include a respective check.

**Comment by julian-smith-artifex-com** [comment]
Fixed in PyMuPDF-1.21.1.
